src/core/lib/layer.js

```diff
@@ -497,11 +497,19 @@ export default class Layer {
   }
 
   diffProps(oldProps, newProps, context) {
+    // Until we have proper prop types, autodetect accessors by querying the attribute manager
+    const {attributeManager} = this.state || {};
+    const propTypes = attributeManager ? attributeManager.getAccessors() : {};
```

Comment:
I assume it is called `propTypes` so that we can expand this object to cover more than functions in the future. If that is true, then it is really strange that `propTypes` comes from a function called `getAccessors` and from the attribute manager.

The logic implemented in this PR will perfectly apply if we just pass the defaultProps object as `propTypes` here. Is it necessary to deduce the list of accessors over and over again?

Reply:
> I assume it is called propTypes so that we can expand this object to cover more than functions in the future. If that is true, then it is really strange that propTypes comes from a function called getAccessors and from the attribute manager.

Changed from `propTypes` to `shallowCompareProps` for now to remove any confusion.

> The logic implemented in this PR will perfectly apply if we just pass the defaultProps object as propTypes here. Is it necessary to deduce the list of accessors over and over again?

The problem here is that @heshan0131 is testing for function changes in the aggregating layers. I cannot just make any function compare shallowly equal until we have resolved that.